src/network/networkd-netlabel.c

```diff
@@ -0,0 +1,190 @@
+        _cleanup_free_ char *addr_str = NULL;
+        int r;
+        const char *label;
+
+        (void) in_addr_prefix_to_string(address->family, &address->in_addr, address->prefixlen, &addr_str);
+
+        SET_FOREACH(label, labels) {
+                r = netlabel_command(NLBL_UNLABEL_C_STATICADD, label, address);
+                if (r < 0)
+                        log_link_warning_errno(address->link, r, "Adding NetLabel %s for IP address %s failed, ignoring",
+                                               label, strna(addr_str));
+                else
+                        log_link_debug(address->link, "Adding NetLabel %s for IP address %s", label, strna(addr_str));
+        }
+}
+
+void address_add_netlabel(const Address *address) {
+        assert(address);
+        assert(address->link);
+
+        if (!address->link->network || !IN_SET(address->family, AF_INET, AF_INET6))
+                return;
+
+        switch (address->source) {
+        case NETWORK_CONFIG_SOURCE_DHCP4:
+                return address_add_netlabel_set(address, address->link->network->dhcp_netlabels);
+        case NETWORK_CONFIG_SOURCE_DHCP6:
+                return address_add_netlabel_set(address, address->link->network->dhcp6_netlabels);
+        case NETWORK_CONFIG_SOURCE_DHCP_PD:
+                return address_add_netlabel_set(address, address->link->network->dhcp_pd_netlabels);
+        case NETWORK_CONFIG_SOURCE_STATIC:
+                return address_add_netlabel_set(address, address->netlabels);
+        default:
+                return;
+        }
+}
+
+void address_del_netlabel(const Address *address) {
+        int r;
+        _cleanup_free_ char *addr_str = NULL;
+
+        assert(address);
+        assert(address->link);
+
+        if (!address->link->network || !IN_SET(address->family, AF_INET, AF_INET6))
+                return;
+
+        (void) in_addr_prefix_to_string(address->family, &address->in_addr, address->prefixlen, &addr_str);
+
+        r = netlabel_command(NLBL_UNLABEL_C_STATICREMOVE, NULL, address);
+        if (r < 0)
+                log_link_warning_errno(address->link, r, "Deleting NetLabels for IP address %s failed, ignoring",
+                                       strna(addr_str));
+        else
+                log_link_debug(address->link, "Deleting NetLabels for IP address %s", strna(addr_str));
+}
+
+int config_parse_netlabel(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        int r;
+        Set **set = data;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(set);
+
+        if (isempty(rvalue)) {
+                *set = set_free_free(*set);
```

Comment:
Please use `set_free()`, as hash_op has free func, i.e., you specified string_hash_ops_free.